shardingsphere-proxy/shardingsphere-proxy-frontend/shardingsphere-proxy-frontend-mysql/src/main/java/org/apache/shardingsphere/proxy/frontend/mysql/command/query/text/query/MySQLMultiStatementsHandler.java

```diff
@@ -0,0 +1,198 @@
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.regex.Pattern;
+
+/**
+ * Handler for MySQL multi statements.
+ */
+public final class MySQLMultiStatementsHandler implements TextProtocolBackendHandler {
+    
+    private static final Pattern MULTI_UPDATE_STATEMENTS = Pattern.compile(";(?=\\s+update)", Pattern.CASE_INSENSITIVE);
+    
+    private static final Pattern MULTI_DELETE_STATEMENTS = Pattern.compile(";(?=\\s+delete)", Pattern.CASE_INSENSITIVE);
+    
+    private final KernelProcessor kernelProcessor = new KernelProcessor();
+    
+    private final JDBCExecutor jdbcExecutor = new JDBCExecutor(BackendExecutorContext.getInstance().getExecutorEngine(), false);
+    
+    private final ConnectionSession connectionSession;
+    
+    private final SQLStatement sqlStatementSample;
+    
+    private final MetaDataContexts metaDataContexts = ProxyContext.getInstance().getContextManager().getMetaDataContexts();
+    
+    private final Map<String, List<ExecutionUnit>> dataSourcesToExecutionUnits = new HashMap<>();
+    
+    private ExecutionContext anyExecutionContext;
+    
+    public MySQLMultiStatementsHandler(final ConnectionSession connectionSession, final SQLStatement sqlStatementSample, final String sql) {
+        this.connectionSession = connectionSession;
+        this.sqlStatementSample = sqlStatementSample;
+        Pattern pattern = sqlStatementSample instanceof UpdateStatement ? MULTI_UPDATE_STATEMENTS : MULTI_DELETE_STATEMENTS;
+        ShardingSphereSQLParserEngine sqlParserEngine = getSQLParserEngine();
+        for (String each : extractMultiStatements(pattern, sql)) {
+            SQLStatement eachSQLStatement = sqlParserEngine.parse(each, false);
+            ExecutionContext executionContext = createExecutionContext(createLogicSQL(each, eachSQLStatement));
+            if (null == anyExecutionContext) {
+                anyExecutionContext = executionContext;
+            }
+            for (ExecutionUnit eachExecutionUnit : executionContext.getExecutionUnits()) {
+                dataSourcesToExecutionUnits.computeIfAbsent(eachExecutionUnit.getDataSourceName(), unused -> new LinkedList<>()).add(eachExecutionUnit);
+            }
+        }
+    }
+    
+    private ShardingSphereSQLParserEngine getSQLParserEngine() {
+        MetaDataContexts metaDataContexts = ProxyContext.getInstance().getContextManager().getMetaDataContexts();
+        return new ShardingSphereSQLParserEngine(DatabaseTypeRegistry.getTrunkDatabaseTypeName(metaDataContexts.getMetaData(connectionSession.getSchemaName()).getResource().getDatabaseType()),
+                metaDataContexts.getGlobalRuleMetaData().findSingleRule(SQLParserRule.class).orElse(null));
+    }
+    
+    private List<String> extractMultiStatements(final Pattern pattern, final String sql) {
```

Comment:
#14636 jdbc still no supports executing multi statement.

Reply:
If you connect Proxy via MySQL JDBC Driver, consider setting rewriteBatchedStatements=true and use the JDBC addBatch & executeBatch, the Driver will handle the SQL concat stuff.
Or consider setting allowMultiQueries=true if you are trying to send multi statements to MySQL.

Reply:
yes, connect url must add allowMultiQueries=true | rewriteBatchedStatements=true
However, what I want to say is that this problem will still exist in shardingsphere-jdbc, we should support this feature by sql parser or satisfies both proxy and jdbc on basic module.

Reply:
This PR is just doing some trick to avoid error occurs in Proxy before the SQL parser supports multi statements. We may following the feature request after 5.1.1 released.

Reply:
> #14636 jdbc still no supports executing multi statement.

Hi @userkdg The exception in this issue may be thrown by JDBC Driver instead of ShardingSphere.